src/renderers/shared/reconciler/ReactMultiChild.js

```diff
@@ -276,8 +292,13 @@ var ReactMultiChild = {
      */
     _updateChildren: function(nextNestedChildrenElements, transaction, context) {
       var prevChildren = this._renderedChildren;
+      var removedChildren = {};
```

Comment:
Right. ReactDOMComponent (in `_updateDOMChildren`) only calls into ReactMultiChild (as `this.updateChildren`) if `this.props.children != null`.